src/cloudant/query.py

```diff
@@ -174,14 +173,7 @@ def __call__(self, **kwargs):
             raise CloudantArgumentError(msg)
 
         # Execute query find
-        headers = {'Content-Type': 'application/json'}
-        resp = self._r_session.post(
-            self.url,
-            headers=headers,
-            data=json.dumps(data, cls=self._encoder)
-        )
-        resp.raise_for_status()
-        return resp.json()
+        return execute_search_or_find_query(self._database, self.url, data)
```

Comment:
Like I mentioned earlier, I'm not a fan of this function but if you're going to use it then I would think that `self._r_session` would need to be removed.  I'm curious as to why pylint did not catch that.  Anyway, don't make any change to this now.  We can discuss this later as well.